source/common/upstream/original_dst_cluster.cc

```diff
@@ -0,0 +1,83 @@
+#include "common/upstream/original_dst_cluster.h"
+
+#include <chrono>
+#include <list>
+#include <string>
+#include <vector>
+
+#include "common/network/address_impl.h"
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+OriginalDstCluster::OriginalDstCluster(const Json::Object& config, Runtime::Loader& runtime,
+                                       Stats::Store& stats,
+                                       Ssl::ContextManager& ssl_context_manager)
+    : ClusterImplBase(config, runtime, stats, ssl_context_manager) {
+  size_t n_hosts = 0;
+  try {
+    std::vector<Json::ObjectSharedPtr> hosts_json = config.getObjectArray("hosts");
+    n_hosts = hosts_json.size();
+  } catch (...) {
+  } // Catch missing hosts config.
```

Comment:
Please don't catch (...) (and definitely don't catch and do nothing). Only specific exceptions. In this case per previous comment I would just use hasObject()